### ISSUE TITLE
Memory problems address book (July 04 emesene development version)

### ISSUE THREAD

**Issue by Sacolevas** [emesene/emesene#1402]
Hello, 

I was testing the latest development version (from emesene github) to test if the segfault was still happening (sometimes  when someone is going to reply to a message) :(

But I found another problem I didn't see earlier. To reproduce, this is what I was doing: I was going to block (or unblock) contacts in my contact list. I have approximately 200 contacts, not a big number; this happen when I repeated the action with about my 30th contact. First, I was going to unblock those contacts and emesene closed unexpectedly and it showed in the windows something about an Address Book memory problem. This is the screenshot:

http://imageshack.us/photo/my-images/546/201207050903151366x768s.png/

I repeated the procedure three times, unclocking contacts, blocking them and unblocking again just to see the same error and being able to put the info in this report. This is what I got, noticing that I felt the debug was behind what I was doing, so I waited about five minutes to see the last messages before closing emesene.

---

1)
<IsBetaMigrated>false</IsBetaMigrated><LastRelevanceUpdate>0001-01-01T00:00:00</LastRelevanceUpdate></abInfo><lastChange>2012-07-05T07:02:25.76-07:00</lastChange><DynamicItemLastChanged>0001-01-01T00:00:00</DynamicItemLastChanged><RecentActivityItemLastChanged>0001-01-01T00:00:00</RecentActivityItemLastChanged><createDate>2011-11-01T15:24:56.19-07:00</createDate><propertiesChanged /></ab></ABFindAllResult></ABFindAllResponse>/soap:Body/soap:Envelope
[09:02:38 INFO papyon.protocol.switchboard] Switchboard timed out. Going to leave it.
[09:02:38 INFO papyon.protocol.switchboard] Leaving switchboard 2105364623084.254111282.632322139
[09:02:38 DEBUG papyon.transport] >>> OUT
[09:02:42 INFO papyon.protocol.switchboard] Switchboard timed out. Going to leave it.
[09:02:42 INFO papyon.protocol.switchboard] Leaving switchboard 2105364623084.254111282.632322139
[09:02:42 DEBUG papyon.transport] >>> OUT
[09:02:42 INFO papyon.protocol.switchboard] Disconnected (2105364623084.254111282.632322139)
[09:02:42 INFO papyon.protocol.switchboard_manager] Closing switchboard handler <SwitchboardP2PTransport peer="a@hotmail.com" guid="xxxxxxxxxxxxxxxxx">

2) 
<MessengerApplicationServiceCreated>false</MessengerApplicationServiceCreated><IsBetaMigrated>false</IsBetaMigrated><LastRelevanceUpdate>0001-01-01T00:00:00</LastRelevanceUpdate></abInfo><lastChange>2012-07-05T07:14:21.54-07:00</lastChange><DynamicItemLastChanged>0001-01-01T00:00:00</DynamicItemLastChanged><RecentActivityItemLastChanged>0001-01-01T00:00:00</RecentActivityItemLastChanged><createDate>2011-11-01T15:24:56.19-07:00</createDate><propertiesChanged /></ab></ABFindAllResult></ABFindAllResponse>/soap:Body/soap:Envelope
[09:14:28 DEBUG papyon.transport] >>> XFR 84 SB
[09:14:35 DEBUG papyon.transport] >>> XFR 85 SB
[09:14:42 DEBUG papyon.transport] >>> XFR 86 SB
[09:14:49 DEBUG papyon.transport] >>> XFR 87 SB
[09:14:55 DEBUG papyon.transport] >>> XFR 88 SB
[09:15:02 DEBUG papyon.transport] >>> XFR 89 SB 

3) 
[09:26:39 DEBUG papyon.transport] >>> OUT
[09:26:39 INFO papyon.protocol.switchboard] Disconnected (1000113137.36155244.1215151)
[09:26:39 INFO papyon.protocol.switchboard_manager] Closing switchboard handler <SwitchboardP2PTransport peer="a@hotmail.com" guid="f763caa73-b622c-454341-84aa-9ad3217g1330">
[09:26:39 INFO papyon.protocol.switchboard] Leaving switchboard 162212030.133113220.5322341
[09:26:39 DEBUG papyon.transport] >>> OUT

---

I hope this helps to solve another problem in this version. By the way, I have seen good differences between this version and the previous one, so I hope those problems with segfaults and memory can be solved for the new release.

Thanks for your attention.

**Comment by moneycat** [comment]
maybe this is similar with #1400

**Comment by Sacolevas** [comment]
I am not sure, because I remember I did something similar (moving contacts from then list of blocked to the list of allowed and viceversa) in version 2.12.5 and something like this never happened. By the way, I tried both ways: blocking and unblocking contacts one by one with right click on mouse and using the menu option of "live messenger" and using the arrows to move from one list to another. In both ways the error eventually appears.

**Comment by c10ud** [comment]
i don't see any error in the debug trace you reported.
however i do see that you got disconnected with an addressbook error
and a "memory error" reason, am i right?

in order to do this did you block/unblocked 50 contacts in a row, if i
read correctly?

2012/7/5 Sacolevas
reply@reply.github.com:

**Comment by Sacolevas** [comment]
Hello, yes, the message is the one that I show in the screenshot. And I don't know if exactly 50, but basically I was manually unblocking contacts consecutively and the error appeared; something that never happened in any previous version (at least, I remember doing something similar in v2.12.5). In order to test if the error was just a coincidence, I reversed the process (blocking) and again, the error appeared. And I repeated the process (unblocking again) and it happened again that the program disconnected by itself and appeared the message you see in the linked image.

**Comment by c10ud** [comment]
you were flooding the server with requests, i think that's normal

**Comment by Sacolevas** [comment]
Yes, but the behavior was strange. I forgot to say the behavior next (I give the steps to reproduce)
- I clicked in the "Close" button that appeared in the image above (next to the Address Book error (Mem ..); by the way, the sentence was not complete; I tried several times with a bigger window and still the message was shown incomplete as in the picture. I assume (Mem ..), means "memory", but there was no way to see the whole announcement. 
- After clicking the button the program didn't close. So I tried to enter again with my username and password.
- I entered, but the program was not working correctly. For example, I tried to access the main menu to see the "Preferences" and it didn't react (nothing was shown). 
- Consequently, I had to disconnect and close the window to start the program again. 

In other words, after the forced disconnection, program should be closed. Otherwise, it won't work in full funcionality after you login again.

Thanks for your attention, I will keep testing :)

**Comment by c10ud** [comment]
it is probably a bug in papyon which is triggered by this weird error, i'll leave this open but i don't think it will ever be fixed

**Comment by c10ud** [comment]
closed as won't fix
